catalog/app/utils/LogicalKeyResolver.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+
+interface S3Handle {
+  bucket: string
+  key: string
+  version?: string
+}
+
+export interface LogicalKeyResolver {
+  (logicalKey: string): S3Handle
```

Comment:
TS guidelines state smth like "prefer interfaces where you can and fall back to types where you can't" and i stick with it (it's also a little easier to add overloads to interface call signature).

> otherwise, it's too easy to confuse it with

well, it's different syntax designed for expressing different things and i dont believe avoiding certain language features bc their syntax _may be_ confused with other features is not right

Reply:
Ok, I didn't know about this guideline

Reply:
it's quite vague tho and i dont know reasoning behind that)